### ISSUE TITLE
Allow search result snippets with file:// protocol (file opened from disk)

### ISSUE THREAD

**Issue by kapooramit** [sphinx-doc/sphinx#10407]
Hi,

I know that this behavior us documented as intended because browsers work don't let this happen: https://github.com/sphinx-doc/sphinx/issues/9156

However, I use another tool called MadCap Flare which generates HTML5 files as output with integrated search index which uses javascript/jquery and it works just fine with file:// as it does with http://.

https://www.madcapsoftware.com/blog/filtering-search-results-in-flare-outputs/

Maybe it is possible here - maybe not. I still think it should be considered because there is significant user base that uses sphinx output right from local disk irrespective of how easy it is to start a local http server.

**Comment by tk0miya** [comment]
If my understanding is correct, it comes from a restriction of web browsers. Sphinx itself does not prohibit it. Actually, searching feature works fine on local browsing w/ Google Chrome.

Please post a pull request if you have ideas to avoid the restriction. Thanks,

**Comment by kapooramit** [comment]
Ok. I will research on how MadCap Flare manages to do it and come back here. By the way, I tried on Chrome too but no text snippet shows up - all I see is the list of pages (clickable links). On localhost or on a web-server, I see the text snippet just fine under each page link.

**Comment by kapooramit** [comment]
Just for the sake of it, let me share that when using search with file:// protocole, I get the following error:
````
Access to XMLHttpRequest at 'file://wsl%24/Ubuntu/home/johndoe/sphinx_rtd_theme/html/support.html' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https.
````

The same error occures multiple times.

**Comment by jbms** [comment]
Yes, you can't use XMLHttpRequest or fetch it with a file:// url.  The search index is loaded using the jsonp method (i.e. via a script tag) which does work with file:// urls.  Currently snippets are generated by fetching the actual html page, which means a separy copy of the page is not required for snippet generation.  Ll

To make snippets work with file://, the text content of every document would need to be additionally stored in jsonp format.  To avoid additional space usage in all cases, this should probably be an option defaulting to false.

**Comment by kapooramit** [comment]
So in other words, it is possible but the option needs to be available so that user/project can set it to True in conf.py or theme using Sphinx. Am I right?

**Comment by jbms** [comment]
Yes it could be possible but would need to be implemented.
